FRC-Extension/RoboRIOCode/RoboRIOConnection.cs

```diff
@@ -26,73 +30,189 @@ public enum ConnectionUser
+        public IPAddress ConnectionIp { get; private set; }
 
-        public ConnectionReturn(ConnectionType type, string ip, bool success)
+        public ConnectionStatus(ConnectionType type, IPAddress ip)
         {
             ConnectionType = type;
             ConnectionIp = ip;
-            Success = success;
         }
     }
 
 
-    public static class RoboRIOConnection
-    {
-
 
+    public class RoboRioConnection : IDisposable
+    {
         public const string RoboRioMdnsFormatString = "roborio-{0}-FRC.local";
+        public const string RoboRioLanFormatString = "roborio-{0}-FRC.lan";
         public const string RoboRioUSBIp = "172.22.11.2";
         public const string RoboRioIpFormatString = "10.{0}.{1}.2";
 
-        private static ConnectionInfo s_adminConnectionInfo;
-        private static ConnectionInfo s_lvUserConnectionInfo;
+        private int m_teamNumber;
+        private IPAddress m_remoteIp;
+        private TimeSpan m_sshTimeout;
+        private ConnectionType m_connectionType = ConnectionType.None;
+
+        private ConnectionInfo m_adminConnectionInfo;
+        private ConnectionInfo m_lvUserConnectionInfo;
+
+        private SshClient m_sshAdminClient;
+        private SshClient m_sshUserClient;
+        private ScpClient m_scpUserClient;
+        private ScpClient m_scpAdminClient;
 
-        public static ConnectionInfo AdminConnectionInfo => s_adminConnectionInfo;
-        public static ConnectionInfo LvUserConnectionInfo => s_lvUserConnectionInfo;
+        public IPAddress IPAddress => m_remoteIp;
 
-        public static async Task<ConnectionReturn> CheckConnectionAsync(string teamNumberS)
+        public ConnectionStatus ConnectionStatus
         {
-            return await CheckConnectionAsync(teamNumberS, TimeSpan.FromSeconds(2)).ConfigureAwait(false);
+            get
+            {
+                if (m_remoteIp == null)
+                {
+                    return null;
+                }
+                return new ConnectionStatus(m_connectionType, m_remoteIp);
+            }
         }
 
-        public static async Task<ConnectionReturn> CheckConnectionAsync(string teamNumberS, TimeSpan timeout)
+        public bool Connected => m_remoteIp != null;
+
+
+        public RoboRioConnection(int teamNumber, TimeSpan sshTimeout)
         {
-            int teamNumber;
-            int.TryParse(teamNumberS, out teamNumber);
+            m_teamNumber = teamNumber;
+            m_sshTimeout = sshTimeout;
+        }
 
+        public void Dispose()
+        {
+            m_scpAdminClient?.Dispose();
+            m_sshUserClient?.Dispose();
+            m_scpUserClient?.Dispose();
+            m_scpAdminClient?.Dispose();
+        }
 
+        public static async Task<RoboRioConnection> StartConnectionTaskAsync(string teamNumberS)
+        {
+            int teamNumber;
+            int.TryParse(teamNumberS, out teamNumber);
             if (teamNumber < 0)
```

Comment:
There are other parts that check if it is 0, but that's a good point. I'll do that.